### PR TITLE
Podman custom port mapping

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes part of #6479 

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: tests/integration/cmd_dev_debug_test.go [hunk 1]
```diff
@@ -40,47 +41,59 @@ var _ = Describe("odo dev debug command tests", func() {
...
-							url := fmt.Sprintf("http://%s", ports[ContainerDebugPort])
-							Expect(url).To(ContainSubstring(LocalDebugPort))
+					By("expecting a ws connection when tried to connect on default debug port locally", func() {
+						// 400 response expected because the endpoint expects a websocket request and we are doing a HTTP GET
+						// We are just using this to validate if nodejs agent is listening on the other side
+						url := fmt.Sprintf("http://%s", ports[ContainerDebugPort])
+						Expect(url).To(ContainSubstring(LocalDebugPort))
+						if !podman {
+							// this check doesn't work for podman unless we use --forward-localhost,
+							// but using it beats the purpose of testing with custom port mapping, so we skip this check
```
Comment (rm3l) [inline tests/integration/cmd_dev_debug_test.go:90]: Not sure I understand why `--forward-localhost` beats the purpose of testing with custom port mapping..  To me, the local port forwarded to the container debug port should be reachable with custom port mapping (if we are running with `--forward-localhost`), and we need to ensure that, no?
`--forward-localhost` is just a way to make port forwarding work on Podman just like on cluster, without having to change the application to expose sensitive ports like debug ones. So it should work with custom port mapping on Podman as well, IMO.

Comment (valaparthvi) [reply tests/integration/cmd_dev_debug_test.go:90]: My bad. I'd misunderstood the way --forward-localhost worked. I'll fix it.

**Comment by valaparthvi** [comment]
OC Failure:
```sh
  << Timeline

  [FAILED] Expected
      <*url.Error | 0xc0004b0900>: {
          Op: "Post",
          URL: "http://127.0.0.1:40605/api/newuser",
          Err: <*errors.errorString | 0xc0001120f0>{s: "EOF"},
      }
  to be nil
  In [It] at: /go/odo_1/tests/e2escenarios/e2e_test.go:319 @ 04/17/23 17:05:29.036
------------------------------
â€¢â€¢â€¢
------------------------------
â€¢ [155.973 seconds]
E2E Test starting with non-empty Directory should verify developer workflow from non-empty Directory
/go/odo_1/tests/e2escenarios/e2e_test.go:193

```

**Comment by valaparthvi** [comment]
/override  OpenShift-Integration-tests/OpenShift-Integration-tests
/override windows-integration-test/Windows-test

**Review by rm3l — APPROVED** [review]

**Pull request merged**